pages/api/post/index.ts

```diff
@@ -66,15 +65,21 @@ async function getCreatablePostFromRequest(
     return null;
   }
 
-  let parsedPrice: any = undefined;
+  let parsedPrice: number | undefined = undefined;
   if (price !== undefined) {
     if (typeof price !== "string") {
       return null;
     }
     parsedPrice = Number.parseInt(price);
-    if (Number.isNaN(parsedPrice) || parsedPrice < 0) {
+    if (
+      parsedPrice === undefined ||
+      Number.isNaN(parsedPrice) ||
+      parsedPrice < 0
```

Comment:
Consider moving this logic to a shared function that both the frontend and backend use

Reply:
They're slightly different, since the frontend has to deal with floats, whereas the backend deals with just integers.